crates/nu-plugin-protocol/src/lib.rs

```diff
@@ -78,25 +78,34 @@ pub enum PipelineDataHeader {
     /// No input
     Empty,
     /// A single value
-    Value(Value),
+    Value {
+        value: Value,
+        metadata: Option<PipelineMetadata>,
+    },
     /// Initiate [`nu_protocol::PipelineData::ListStream`].
     ///
     /// Items are sent via [`StreamData`]
-    ListStream(ListStreamInfo),
+    ListStream {
```

Comment:
Was there any reason why you decided against adding these to `ListStreamInfo` / `ByteStreamInfo`? I have those as separate structs mostly to facilitate passing them around since there's no way to type a particular enum variant, but I don't remember if there's some reason you wouldn't want to have metadata in there.

Reply:
Because of Value variant. If there was a ValueInfo for instance, I would've went that route. It seemed weird to have pipeline metadata at the top level for value, but under info for list stream and byte stream.

Reply:
I think if you'd like to see it be perfectly orthogonal then it's fine to create a `ValueInfo`. I'm just also thinking of compactness for the wire format, there's not really any reason to separate out `info`

Reply:
OK, i put the metadata inside info for bytestream and liststream and leave it top level for Value.